src/IB/IBHydrodynamicSurfaceForceEvaluator.cpp

```diff
@@ -336,25 +409,73 @@ IBHydrodynamicSurfaceForceEvaluator::computeHydrodynamicForce()
     }
 
     return;
+} //  computeHydrodynamicForce
 
-} // computeHydrodynamicForce
+void
+IBHydrodynamicSurfaceForceEvaluator::setSurfaceContourLevel(double s)
+{
+    d_surface_contour_value = s;
+    return;
+} // setSurfaceContourLevel
+
+void
+IBHydrodynamicSurfaceForceEvaluator::writeToFile(bool write_to_file)
+{
+    d_write_to_file = write_to_file;
+
+    // Set up the streams for printing force and torque
+    if (d_write_to_file && SAMRAI_MPI::getRank() == 0)
+    {
+        std::string force;
+        force = "Hydro_Force_" + d_ls_solid_var->getName();
+        bool from_restart = RestartManager::getManager()->isFromRestart();
+        if (from_restart)
+        {
+            d_hydro_force_stream.reset(new std::ofstream(force.c_str(), std::fstream::app));
+            d_hydro_force_stream->precision(10);
+        }
+        else
+        {
+            d_hydro_force_stream.reset(new std::ofstream(force.c_str(), std::fstream::out));
+            d_hydro_force_stream->precision(10);
+        }
+
+        std::string torque;
+        torque = "Hydro_Torque_" + d_ls_solid_var->getName();
+        if (from_restart)
+        {
+            d_hydro_torque_stream.reset(new std::ofstream(torque.c_str(), std::fstream::app));
+            d_hydro_torque_stream->precision(10);
+        }
+        else
+        {
+            d_hydro_torque_stream.reset(new std::ofstream(torque.c_str(), std::fstream::out));
+            d_hydro_torque_stream->precision(10);
+        }
+    }
+    return;
+} // writeToFile
 
 /////////////////////////////// PROTECTED ////////////////////////////////////
 
 /////////////////////////////// PRIVATE //////////////////////////////////////
 
 void
 IBHydrodynamicSurfaceForceEvaluator::fillPatchData(Pointer<PatchHierarchy<NDIM> > patch_hierarchy,
-                                                   const double fill_time)
+                                                   const double fill_time,
+                                                   bool use_current_ctx,
+                                                   bool use_new_ctx)
```

Comment:
We should make the arguments to this the same as `computeHydrodynamicForceTorque`: it has the same logic for determining which context to use.

We should also log that this improvement (i.e., we no longer assume that we are at the current time) is incompatible.